### PR TITLE
Per instance ReactChoreographer accessor

### PR DESCRIPTION
**Pull request by reseul**
An accessor that can be used to access a ReactChoreographer pertaining to a particular top level window.
Can be used to hook custom animations/etc into the correct choreographer.

### CODE REVIEW

File: ReactWindows/ReactNative.Shared/Modules/Core/ReactChoreographer.cs [hunk 1]
```diff
@@ -82,7 +116,7 @@ private ReactChoreographer(CoreApplicationView applicationView)
         public event EventHandler<FrameEventArgs> IdleCallback;
 
         /// <summary>
-        /// The choreographer instance.
+        /// The choreographer main instance.
         /// </summary>
         public static IReactChoreographer Instance
```
Comment (rozele) [inline ReactWindows/ReactNative.Shared/Modules/Core/ReactChoreographer.cs]: Do we need this special case? Is it more developer-friendly to have this "main" `Instance` and an explicit alt `ThreadInstance`, or should all calls to `Instance` be specific to the thread?

File: ReactWindows/ReactNative.Shared/Modules/Core/ReactChoreographer.cs [hunk 2]
```diff
@@ -55,9 +58,40 @@ private ReactChoreographer(CoreApplicationView applicationView)
...
+                return instance;
+            }
         }
 #else
         private ReactChoreographer() { }
+
+        /// <summary>
+        /// The choreographer instance (returns same as Instance for WPF)
+        /// </summary>
+       public static IReactChoreographer ThreadInstance => s_instance;
```
Comment (rozele) [inline ReactWindows/ReactNative.Shared/Modules/Core/ReactChoreographer.cs]: Do we need `CreateSecondaryInstance` anymore?

File: ReactWindows/ReactNative.Shared/Modules/Core/ReactChoreographer.cs (hunk 1 above)
Comment (reseul) [reply ReactWindows/ReactNative.Shared/Modules/Core/ReactChoreographer.cs]: I think I know where you're headed, but ReactChoreographer.Instance is called from random threads (module one, etc.), making any thread affinity dangerous.

---
In reply to: [225740827](https://github.com/Microsoft/react-native-windows/pull/2000#discussion_r225740827) [](ancestors = 225740827)

File: ReactWindows/ReactNative.Shared/Modules/Core/ReactChoreographer.cs (hunk 2 above)
Comment (reseul) [reply ReactWindows/ReactNative.Shared/Modules/Core/ReactChoreographer.cs]: As opposed to calling ThreadInstance to construct it? IF we still keep Instance and ThreadInstance separate, would you still want ThreadInstance to create the instance (whereas Instance wouldn't), or not? I can go either way.

---
In reply to: [225741039](https://github.com/Microsoft/react-native-windows/pull/2000#discussion_r225741039) [](ancestors = 225741039)

Comment (rozele) [reply ReactWindows/ReactNative.Shared/Modules/Core/ReactChoreographer.cs]: Correct - saying that CreateSecondaryInstance is more or less the same as ThreadInstance and can be removed, correct?

Comment (rozele) [reply ReactWindows/ReactNative.Shared/Modules/Core/ReactChoreographer.cs]: Seems like ThreadInstance could just lazily create the instance. Any risk to that?

Comment (reseul) [reply ReactWindows/ReactNative.Shared/Modules/Core/ReactChoreographer.cs]: Ok, will do lazy creation for the ThreadInstance.

Comment (reseul) [reply ReactWindows/ReactNative.Shared/Modules/Core/ReactChoreographer.cs]: @rozele ah, I take that back, it's not that easy..
The existing usages of secondary choreographers are inside UIViewOperationQueueInstance where:
- constructor is called from LM thread 
- a bunch of choreographer method calls (ActivateCallback, etc.) are from other threads 
So I don't see any "good" place where I can hook a "_reactChoregrapher = ReactChoreographer.ThreadInstance"..

**Pull request merged**